syntax = "proto3";

package e2e_message;

message RuleCheckRequest {
    bytes taskhash = 1;
}

message RuleCheckResult {
    bool status = 1;
    bytes task_result = 2; 
    string error_info = 3; // empty if status is true
}

message RuleCheckResponse {
    bytes taskhash = 1;  
    RuleCheckResult result =2;  
    bytes rule_file_hash = 3;
    bytes signature = 4;
}